frontend/src/features/housingCompany/HousingCompanyDetailsPage.tsx

```diff
@@ -1,23 +1,96 @@
-import {Button, IconPlus, StatusLabel, Tabs} from "hds-react";
+import {Button, Dialog, IconPlus, StatusLabel, Tabs} from "hds-react";
 import {Link, useParams} from "react-router-dom";
 
-import {useGetHousingCompanyDetailQuery} from "../../app/services";
-import {DetailField, EditButton, Heading, ImprovementsTable, QueryStateHandler} from "../../common/components";
+import {useState} from "react";
+import {useForm} from "react-hook-form";
+import {
+    useCreateFromSalesCatalogMutation,
+    useGetHousingCompanyDetailQuery,
+    useValidateSalesCatalogMutation,
+} from "../../app/services";
+import {
+    CloseButton,
+    DetailField,
+    EditButton,
+    Heading,
+    ImprovementsTable,
+    QueryStateHandler,
+    SaveButton,
+} from "../../common/components";
+import {FileInput} from "../../common/components/form";
 import {getHousingCompanyHitasTypeName, getHousingCompanyRegulationStatusName} from "../../common/localisation";
-import {IHousingCompanyDetails} from "../../common/schemas";
-import {formatAddress, formatDate, formatMoney} from "../../common/utils";
+import {IHousingCompanyDetails, ISalesCatalogApartment} from "../../common/schemas";
+import {formatAddress, formatDate, formatMoney, hdsToast} from "../../common/utils";
 import {HousingCompanyApartmentResultsList} from "../apartment/ApartmentListPage";
 
 const LoadedHousingCompanyDetails = ({housingCompany}: {housingCompany: IHousingCompanyDetails}) => {
+    const [isImportModalOpen, setIsImportModalOpen] = useState(false);
     const params = useParams() as {readonly housingCompanyId: string};
+    const salesCatalogForm = useForm({defaultValues: {salesCatalog: null}});
+    const [validateSalesCatalog, {data: validateData, isLoading: isValidating, error: validateError}] =
+        useValidateSalesCatalogMutation();
+    const [createImportedApartments, {error: createError}] = useCreateFromSalesCatalogMutation();
+    const {handleSubmit} = salesCatalogForm;
+    const handleCreateButton = () => {
+        const importedApartments: object[] = [];
+        validateData.apartments.forEach((apartment: ISalesCatalogApartment) => {
+            importedApartments.push({
+                stair: apartment.stair,
+                floor: apartment.floor,
+                apartment_number: apartment.apartment_number,
+                rooms: apartment.rooms,
+                apartment_type: (apartment.apartment_type as {id: string}).id,
+                surface_area: apartment.surface_area,
+                share_number_start: apartment.share_number_start,
+                share_number_end: apartment.share_number_end,
+                catalog_purchase_price: apartment.catalog_purchase_price,
+                catalog_primary_loan_amount: apartment.catalog_primary_loan_amount,
+            });
+        });
+        console.log("importedApartments:", importedApartments);
```

Comment:
`console.log`s can probably be removed, and `// eslint-disable-next-line no-console` added to `console.warn`s 😉